src/server.cpp

```diff
@@ -411,41 +454,72 @@ static int statusToExitCode(const Status& status) {
     return OVMS_EX_FAILURE;
 }
 
-// OVMS Start
-int Server::start(int argc, char** argv) {
-    installSignalHandlers();
-    int result = OVMS_EX_OK;
-
+std::variant<std::pair<ServerSettingsImpl, ModelsSettingsImpl>, std::pair<int, std::string>> Server::parseArgs(int argc, char** argv) {
     try {
         CLIParser parser;
         ServerSettingsImpl serverSettings;
         ModelsSettingsImpl modelsSettings;
-        parser.parse(argc, argv);
+        auto successOrExit = parser.parse(argc, argv);
+        // Check for error in parsing
+        if (!std::holds_alternative<bool>(successOrExit)) {
+            auto printAndExit = std::get<std::pair<int, std::string>>(successOrExit);
+            return printAndExit;
+        }
         parser.prepare(&serverSettings, &modelsSettings);
+        return std::make_pair(serverSettings, modelsSettings);
+    } catch (const std::exception& e) {
+        return std::make_pair(OVMS_EX_USAGE, e.what());
+    }
+}
+
+int Server::startServerFromSettings(ServerSettingsImpl& serverSettings, ModelsSettingsImpl& modelsSettings) {
+    installSignalHandlers();
+    int result = OVMS_EX_OK;
+    Server::setExitStatus(0);
 
-        Status ret = start(&serverSettings, &modelsSettings);
+    try {
+        Status ret = startFromSettings(&serverSettings, &modelsSettings);
         ModulesShutdownGuard shutdownGuard(*this);
         if (!ret.ok()) {
             return statusToExitCode(ret);
         }
-        while (!shutdown_request &&
+        while (!getShutdownStatus() &&
                (serverSettings.serverMode == HF_PULL_AND_START_MODE || serverSettings.serverMode == SERVING_MODELS_MODE)) {
             std::this_thread::sleep_for(std::chrono::milliseconds(200));
         }
-        if (shutdown_request == 2) {
+        if (getShutdownStatus() == 2) {
             SPDLOG_ERROR("Illegal operation. OVMS started on unsupported device");
         }
     } catch (const std::exception& e) {
         SPDLOG_ERROR("Exception; {}", e.what());
         result = OVMS_EX_FAILURE;
+        Server::setExitStatus(1);
         return result;
     }
 
+    Server::setExitStatus(1);
     return EXIT_SUCCESS;
 }
 
+// OVMS Start
+int Server::start(int argc, char** argv) {
+    auto paramsOrExit = parseArgs(argc, argv);
```

Comment:
Ideally this could be decoupled with Server:
we would have in main:
```
CLIParser parser;
parser.parse(argc,argv)
```
Then
```
server.start(serverSettings, modelsSettings)
```